checkST2110.js

```diff
@@ -1073,108 +1305,317 @@ const test_21_82_2 = (sdp, params) => {
 };
 
 const section_30_62 = (sdp, params) => {
-  let tests = [ test_30_62_1, test_30_62_2, test_30_62_3, test_30_62_4, test_30_62_5 ];
+  let tests = [test_30_62_1, test_30_62_2, test_30_62_3, test_30_62_4, test_30_62_5];
   return concat(tests.map(t => t(sdp, params)));
 };
 
 const section_21_81 = (sdp, params) => {
-  let tests = [ test_21_81_1, test_21_81_2 ];
+  let tests = [test_21_81_1, test_21_81_2];
   return concat(tests.map(t => t(sdp, params)));
 };
 
 const section_21_82 = (sdp, params) => {
-  let tests = [ test_21_82_1, test_21_82_2 ];
+  let tests = [test_21_82_1, test_21_82_2];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_53 = (sdp, params) => {
+  let tests = [test_22_53_1];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_6 = (sdp, params) => {
+  let tests = [test_22_6];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_71 = (sdp, params) => {
+  let tests = [test_22_71_1];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_72 = (sdp, params) => {
+  let tests = [test_22_72_1];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_73 = (sdp, params) => {
+  let tests = [test_22_73_1];
+  return concat(tests.map(t => t(sdp, params)));
+};
+
+const section_22_74 = (sdp, params) => {
+  let tests = [test_22_74_1];
   return concat(tests.map(t => t(sdp, params)));
 };
 
 // Test ST2110-10 Appendix B Test 1 - Check that the SDP file given is not a straight copy
-const no_copy = sdp => {
+const no_copy = (sdp, specSDP) => {
   let lines = splitLines(sdp.trim());
-  let exlines = splitLines(specExample);
+  let exlines = splitLines(specSDP);
   let length = lines.length < exlines.length ? lines.length : exlines.length;
   let matching = true;
-  for ( let x = 0 ; x < length ; x++ ) {
+  for (let x = 0; x < length; x++) {
     if (!lines[x].replace(/\s+/g, ' ').startsWith(exlines[x])) {
       matching = false;
       break;
     }
   }
-  return matching ? [ new Error(
+  return matching ? [new Error(
     'SDP file given is a fairly obvious copy of the example in the standard.')] :
     [];
 };
 
+// Test ST2110-10 Appendix B Test 1 - Check that the SDP file given is not a straight copy
+const no_copy_20 = sdp => {
+  return no_copy(sdp, specExample20);
+};
+
+// Test TR-08 Appendix 1 Test 1 - Check that the SDP file given is not a straight copy
+const no_copy_22 = sdp => {
+  return no_copy(sdp, specExample22);
+};
+
 const allSections = (sdp, params) => {
-  let sections = [
-    section_10_74, section_10_81, section_10_82, section_10_83,
-    section_20_71, section_20_72, section_20_73, section_20_74,
-    section_20_75, section_20_76, section_30_62,
-    section_21_81, section_21_82  ];
-  if (params.noCopy) {
-    sections.push(no_copy);
+  // Declare the array holding test functions
+  let sections = [];
+  // Pull out the media type
+  let [mtParams, errors] = extractMTParams(sdp, params);
```

Comment:
:clap: I think this is a very nice idea!! SDPoker can do some very basic (RFC 4566) tests even if there are no `m=` lines, but for the ST 2110 tests we can need to find one. Let's limit the ST 2110-30 tests appropriately only to the appropriate media subtypes too.

Reply:
good deal. Will resolve this item.